### ISSUE TITLE
CaseInsensitiveEnumArgumentType - Binary incompatible change compared to 0.7.0

### ISSUE THREAD

**Issue by jstuyts** [tatsuhiro-t/argparse4j#99]
By using [japicmp](siom79.github.io/japicmp/) binary incompatibilities were detected between the current codebase and the last released version 0.7.0. It only affects 1 class: `CaseInsensitiveEnumArgumentType`. This is the list of reported problems:

* The class went from being concrete to being abstract.
* The constructor went from being public to being protected.

We either have to fix this, or increase the version of the next release to 1.0.0. If we choose the latter, we might want to do further restructuring to make clear what the supported public API is.

**Comment by tatsuhiro-t** [comment]
CaseInsensitiveEnumArgumentType has been introduced after 0.7.0 release, so I'm not sure why it is picked up by binary incompatibility detector.  Does it affect other subclasses?

https://github.com/tatsuhiro-t/argparse4j/tree/974ae57d6d3453da4168633d8a84dc310b96bbbf/src/main/java/net/sourceforge/argparse4j/impl/type

**Comment by jstuyts** [comment]
I already thought it was strange that the Git history was so short.

No, it only reports binary incompatibility for that class. Maybe I am not using japicmp right, but there really is not a lot you can do wrong. I will compare 0.7.0 to 0.8.0-SNAPSHOT myself and see if it matches what japicmp is reporting.

**Comment by jstuyts** [comment]
`CaseInsensitiveEnumArgumentType` was already present in 0.7.0, so the reporting is correct.

During manual comparison I noticed the following:

* New abstract methods in `CaseInsensitiveEnumArgumentType` were not reported. Although sub  classes will load, invocation of one of these methods will throw an exception. I will try to get these methods reported too, but it looks like this is not possible. This also applies to:
    * Interface `ArgumentParser`
    * Interface `Subparser`
* `HelpScreenException` has been moved, but I think it was always part of the API, so the move will break existing clients. It was not reported because I exclude the whole internal package. I will enable reporting for this class, and create a separate issue for fixing it.
